rocks-lib/src/rockspec/build/builtin.rs

```diff
@@ -54,6 +87,139 @@ pub struct ModulePaths {
     pub libdirs: Vec<PathBuf>,
 }
 
+impl Build for BuiltinBuildSpec {
+    fn run(self, _rockspec: Rockspec, output_paths: TreeLayout, _no_install: bool) -> Result<()> {
+        // Detect all Lua modules
+        let modules = autodetect_modules()?
+            .into_iter()
+            .chain(self.modules)
+            .collect::<HashMap<_, _>>();
+
+        fn lua_module_to_pathbuf(path: &str, extension: &str) -> PathBuf {
+            PathBuf::from(path.replace('.', std::path::MAIN_SEPARATOR_STR) + extension)
+        }
+
+        for (destination_path, module_type) in &modules {
+            match module_type {
+                ModuleSpec::SourcePath(source) => {
+                    let destination_path = lua_module_to_pathbuf(destination_path, ".lua");
+
+                    let target = output_paths.src.join(destination_path);
+
+                    std::fs::create_dir_all(target.parent().unwrap())?;
+
+                    std::fs::copy(source, target)?;
+                }
+                ModuleSpec::SourcePaths(files) => {
+                    let destination_path =
+                        lua_module_to_pathbuf(destination_path, std::env::consts::DLL_SUFFIX);
+                    let target = output_paths.lib.join(destination_path);
+
+                    let parent = target.parent().expect("TODO");
+                    let file = target.file_name().expect("TODO");
+
+                    std::fs::create_dir_all(parent)?;
+
+                    cc::Build::new()
```

Comment:
question: do we need to add lua's `include/` here?

suggestion: we should add a test that builds a simple lua C library and make sure we can use it with lua.

Reply:
Great question... when testing with lua-cjson, there were no compilation issues. We should definitely test on a range of different rockspecs, since I can indeed imagine that this may cause problems if not implemented.

Reply:
You know what, I'm going to add a TODO to add behaviour for this, but for now I'd say we don't need it. It's a very complex problem to solve (even with all the helper crates for compiling lua), so much so that it'd suck up days (maybe weeks) of development time. Let's postpone this and work on other aspects of the codebase :)